File: src/06.ts
import {makeCharNGram} from './05'

const unionSet = (setA: Set<string>, setB: Set<string>): Set<string> => {
    return new Set([...setA, ...setB])
}

const diffSet = (setA: Set<string>, setB: Set<string>): Set<string> => {
    return new Set([...setA].filter(a => !setB.has(a)))
}

const intersectSet = (setA: Set<string>, setB: Set<string>): Set<string> => {
    return new Set([...setA].filter(a => setB.has(a)))
}

{
    const setA = new Set(makeCharNGram(2, 'paraparaparadise'))
    const setB = new Set(makeCharNGram(2, 'paragraph'))
    console.log(unionSet(setA, setB))
    console.log(diffSet(setA, setB))
    console.log(intersectSet(setA, setB))
    console.log(intersectSet(setA, setB).has('se'))
}

File: src/06.py
from share.make_n_gram import make_char_n_gram

set_a = set(make_char_n_gram(2, "paraparaparadise"))
set_b = set(make_char_n_gram(2, "paragraph"))
print(set_a | set_b)
print(set_a - set_b)
print(set_a & set_b)
print("se" in (set_a & set_b))
